CRM/Contact/Page/View/Summary.php

```diff
@@ -310,6 +311,18 @@ public function view() {
           'count' => CRM_Contact_BAO_Contact::getCountComponent($u, $this->_contactId),
           'class' => 'livePage',
         );
+        // Add "Related Cases" tab
+        if($i == 'case' && $entityType == 'Organization' && Civi::settings()->get('civicaseRelatedCasesTab')) {
+          $relIds = CRM_Case_BAO_Case::getOrganizationRelatedCaseIds($this->_contactId);
+          $allTabs[] = array(
+            'id' => 'relatedcases',
+            'url' => CRM_Utils_System::url("civicrm/contact/view/$u", $q."&relatedCases=1"),
+            'title' => 'Related Cases',
+            'weight' => $elem['weight']+1,
+            'count' => CRM_Contact_BAO_Contact::getCountComponent($u, $relIds),
```

Comment:
I don't thiink this would work, and if it does, it probably shouldn't. If I'm not mistaken, $u at this point is 'case'. Calling getCountComponent with this first parameter will try to count the number of cases for the ID of the contact you pass in the second parameter. Furthermore that second parameter is expected to be a single integer, and you're passing an array. So, basically, calling getCountComponent is used to obtain the count for stuff related to a single contact.  

Maybe it could work, as at the end the counting is being done by an API call, and I think the API is capable of handling the array of contact ID's, but I think it'd be kind of abusing of the knowledge we have of how that's implemented, so it definitely increases coupling. If in the future this is changed, or input parameter is enforced to be a single integer, the new tab would break. If we really wanted to keep using getCountComponent, at the very least we'd have to update the documentation for it and the code so it is clear that $contactID can be an array of ID's, but I don't think it's worth it. Or we could pass the organization's ID, and send 'relatedcases' as the first parameter and implement within getCountComponen something to handle the count. The simplest would be to try something like:
```php
  'count' => CRM_Case_BAO_Case::getOrganizationRelatedCasesCount($this->_contactId),
```

And just see if it flies with core team.